src/client_lib/pegasus_client_factory_impl.cpp

```diff
@@ -23,7 +23,7 @@ bool pegasus_client_factory_impl::initialize(const char *config_file)
             // use config file to run
             char exe[] = "client";
             char config[1024];
-            sprintf(config, "%s", config_file);
+            snprintf(config, 1024, "%s", config_file);
```

Comment:
I think you should define a const variable for the value `1024`

Reply:
That's true, but the size of these buffers are already hard-coded and you have plans to refactor them afterwards with maybe fmtlib, so I don't want to bother to introduce new size constants for each of them. Time would be better spent on refactoring with fmtlib.

P.S. In Pegasus' code base, there are at least 25 `char` arrays with hard-coded size and no comment explaining the choice of different sizes. That should be improved in other PRs.